src/globus_sdk/login_flows/command_line_login_flow_manager.py

```diff
@@ -128,4 +137,16 @@ def prompt_for_code(self) -> str:
         :returns: The authorization code entered by the user.
         """
         code_prompt = "Enter the resulting Authorization Code here: "
-        return input(code_prompt).strip()
+        with self.handle_input_errors():
+            return input(code_prompt).strip()
+
+    @contextmanager
+    def handle_input_errors(self) -> t.Iterator[None]:
```

Comment:
Two notes:

- Because this is a public interface, it should have a docstring which tells users what it does
- Should it be named `handle_input_errors` or something specific to the prompt step, i.e. `handle_prompt_errors`?

Under that (renaming) interpretation, combined with the potential rescoping I mentioned above, we'd get something like...

```python
@contextlib.contextmanager
def handle_prompt_errors(self) -> t.Iterator[None]:
    """
    A contextmanager which wraps the ``prompt_for_input`` step when a login flow is run, and may
    catch and reshape errors emitted by that method.

    The default implementation converts ``EOFError``\s to ``CommandLineLoginFlowEOFError``\s.
    """
```